src/bot.js

```diff
@@ -108,6 +108,14 @@ const checkBlock = (oldBlock, newBlock) => {
 }
 
 const connect = () => {
+
+    // close existing viewer instance if the bot reconnects to avoid EADDRINUSE
+    if (bot != null) {
+        if (bot.viewer != null) {
```

Comment:
The viewer is technically undefined but it's `undefined == null` so it works. Do you want to keep this as is? If so you can merge

Reply:
The universal solution for checking (variable !== undefined && variable !== null) in JS is (variable != null) so I think we should leave it this way. I also replaced (variable !== undefined && variable !== null) with (variable != null) in other places in bot.js in previous commits.

Reply:
Alright, TIL.